avocado/formatters.py

```diff
@@ -95,139 +105,238 @@ def to_FORMAT(value, field=field, concept=self.concept,
 
-        # Create list of formats to attempt for multi-value processing.
-        multi_formats = list(preferred_formats)
+    def __unicode__(self):
+        return u'{0}'.format(self.name or self.__class__.__name__)
+
+    def _process(self, values, kwargs=None):
+        # Create a record of the values.
+        record = self._recordclass(*values)
 
-        # Append concept type as a format if one is defined.
-        if self.concept and self.concept.type:
-            multi_formats.append(self.concept.type)
+        # Process multi-value format methods first.
+        for method in self.multi_formats:
+            output = self._process_multiple(method,
+                                            value=record,
+                                            kwargs=kwargs)
 
-        multi_formats.append('raw')
+            if output is not skip:
+                return output
 
-        # Iterate over all preferred formats and attempt to process the values.
-        # Formatter methods that process all values must be tracked and
-        # attempted only once. They are removed from the list once attempted.
-        # If no multi-value methods succeed, each value is processed
-        # independently with the remaining formats.
-        for f in multi_formats:
-            method = getattr(self, 'to_{0}'.format(f), None)
+        # Optimization. Return raw values if no single-value formatters
+        # are being used.
+        if not self.single_formats:
+            return tuple(record)
 
-            # This formatter does not support this format, remove it
-            # from the available list.
-            if not method:
-                continue
+        # Process each value separately with the corresponding field.
+        output = []
 
-            if getattr(method, 'process_multiple', False):
-                try:
-                    output = method(values,
-                                    fields=self.fields,
-                                    concept=self.concept,
-                                    process_multiple=True,
-                                    **context)
+        for i, key in enumerate(record._fields):
+            field = self.fields.get(key)
 
-                    if not isinstance(output, dict):
-                        return OrderedDict([(self.concept.name, output)])
+            value = record[i]
 
-                    return output
+            for method in self.single_formats:
+                value = self._process_single(method,
+                                             value=value,
+                                             field=field,
+                                             kwargs=kwargs)
 
-                except Exception:
-                    if self.concept and self.concept not in self._errors:
-                        self._errors.add(self.concept)
-                        log.exception('Multi-value formatter error')
+                if value is skip:
```

Comment:
I think this could be reduced to:

```
if value is not skip:
    output.extend(value)
    break
```

Using that, the `continue` would be free as it would just be the next loop iteration. Reduces lines a bit and removes a branch from the loop.